src/main/java/frc/robot/subsystems/DriveSystem.java

```diff
@@ -361,6 +360,52 @@ public CommandBase rotateToAngle(Rotation2d angle) {
     );
   }
 
+  public CommandBase autoBalance() {
+    
+    return runEnd(
+      // runs repeatedly until end of command
+      () -> {
+        // radians
+        double currAngle = Math.toRadians(gyro.getRoll());
+
+        double endAngle = 0;
+
+        // rad/s ????
+        double nextVel = balanceController.calculate(currAngle, endAngle);
+
+        // convert radial velocity to drivetrain speeds
+        ChassisSpeeds balanceVel = new ChassisSpeeds(nextVel, 0, 0);
+        DifferentialDriveWheelSpeeds wheelSpeeds = kinematics.toWheelSpeeds(balanceVel);
+
+        setDrivePIDControllers(wheelSpeeds);
+      },
+      // runs once at end of command 
+      () -> {
+        balanceController.reset();
+        
+        // stop motors
+        leftController.setReference(0, ControlType.kVelocity);
+        rightController.setReference(0, ControlType.kVelocity);
+        
+        frontLeft.stopMotor();
+        frontRight.stopMotor();
+      }
+    ).until(
+      // returns true if robot is at end angle
+      balanceController::atSetpoint
+    );
```

Comment:
this might need a different end condition: shouldn't it keep running even after it balances the first time in case of disruptions like other robots?